### PR TITLE
Feature/custom nfts

### PR DESCRIPTION
**Pull request by Cancuuu**
## ADD CUSTOM NFTS FEATURE
ISSUE NUMBER #559 

- Added buttons on home page
- Added GET_NFT_INFO method and created in plug-controller too
- Added ADD_CUSTOM_NFT method and created in plug-controller too
- New add custom nft components and pages created

### Type of changes included:

- [x] Components
- [ ] Business Logic
- [ ] Bug fixes
- [x] Styling
- [x] Code Refactor

### Screenshots/Gifs:

image
image
image

### Notes:

*Place special notes here*

### Tested on:

- [x] Chrome
- [ ] Firefox
- [ ] Safari

### CODE REVIEW

**Comment by riveign** [comment]
@Cancuuu question, what should we do with the thumbnail image?

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@metamask/post-message-stream": "^4.0.0",
     "@psychedelic/browser-rpc": "2.1.0",
     "@psychedelic/dab-js": "1.4.5",
-    "@psychedelic/plug-controller": "0.19.7",
+    "@psychedelic/plug-controller": "0.19.8",
```
Comment (angarita-dev) [inline package.json]: We should use version `0.20`

File: source/Background/Keyring/index.js [hunk 2]
```diff
@@ -202,10 +206,12 @@ export const getKeyringHandler = (type, keyring) => ({
     return parsed;
   },
   [HANDLER_TYPES.GET_ASSETS]: async ({ refresh }) => {
+    console.log('se activa get assets!');
```
Comment (angarita-dev) [inline source/Background/Keyring/index.js]: Remove debugging `console.log`

File: source/Background/Keyring/index.js [hunk 3]
```diff
@@ -202,10 +206,12 @@ export const getKeyringHandler = (type, keyring) => ({
...
     try {
       if (!keyring?.isUnlocked) return {};
 
       const { wallets, currentWalletId } = await keyring.getState();
+      console.log('el keyring getState me devuelve', wallets, currentWalletId);
```
Comment (angarita-dev) [inline source/Background/Keyring/index.js]: same ^

File: source/Background/Keyring/index.js [hunk 4]
```diff
@@ -279,6 +285,34 @@ export const getKeyringHandler = (type, keyring) => ({
...
+        return { error: e.message };
+      }
+    },
+  [HANDLER_TYPES.ADD_CUSTOM_NFT]:
+    async ({ canisterId, standard }) => {
+      try {
+        const nfts = await keyring.registerNFT({
+          canisterId, standard,
+        });
+        console.log('desde plug luego de llamar a registerNFT del keyring ->', nfts);
```
Comment (angarita-dev) [inline source/Background/Keyring/index.js]: ^

File: source/components/Tokens/components/TokenSelector/index.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    <>
+      <div
+        onClick={() => navigator.navigate('add-token')}
+        className={classes.addTokenButton}
+        data-testid="add-custom-token-button"
+      >
+        <img
+          src={coins}
+          alt="reload"
+        />  <Typography variant="h5" data-testid="wallet-name">Add Token</Typography>
```
Comment (angarita-dev) [inline source/components/Tokens/components/TokenSelector/index.jsx]: Maybe we could add this text to `i18n`

File: source/components/Tokens/index.jsx [hunk 6]
```diff
@@ -20,8 +19,8 @@ const Tokens = () => {
   const [loading, setLoading] = useState(true);
   const dispatch = useDispatch();
   const icpPrice = useICPPrice();
-  const { navigator } = useRouter();
   const { onScroll, fullScroll } = useScroll();
+  const [activateButton, setActivateButton] = useState(false);
```
Comment (angarita-dev) [inline source/components/Tokens/index.jsx]: We should probably change this to `displayCustomToken` or something more descriptive

File: source/views/Extension/Views/AddNFT/Steps/Step2.jsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+import useStyles from '../styles';
+
+
+const Step2 = ({ selectedToken, handleClose }) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const [loading, setLoading] = useState(false);
+  // const displayToken = parseTokenBySymbol(selectedToken); // podemos agregar un parser para terminar con el ultimo objeto
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/Steps/Step2.jsx]: remove comment

File: source/views/Extension/Views/AddNFT/Steps/Step2.jsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+
+  const { principalId } = useSelector((state) => state.wallet);
+
+  const registerToken = () => {
+    setLoading(true);
+    console.log('el selected token es ->', selectedToken);
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/Steps/Step2.jsx]: console.log

File: source/views/Extension/Views/AddNFT/components/CustomNFT.jsx [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+  };
+
+  const handleCloseDialog = (value) => {
+    setStandard(value?.name ?? standard);
+    setDialogOpen(false);
+    setTokenError('');
+  };
+
+  const isInvalidInterfaceError = tokenError.includes('Call failed:');
+  const isInvalidCanisterError = tokenError === 'The provided canister id is invalid';
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/components/CustomNFT.jsx:68]: maybe we could move this to an object? or a helper function

File: source/views/Extension/Views/AddNFT/components/CustomNFT.jsx [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+        {tokenError && (
+          <Grid item xs={12}>
+            <div className={classes.appearAnimation}>
+              <Alert
+                type="danger"
+                title={(
+                  <div>
+                    <span data-testid="token-error">
+                      {isInvalidCanisterError
+                        ? t('addNFT.invalidCanisterTokenError')
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/components/CustomNFT.jsx:125]: yeah maybe using an object we could get the translation string directly

**Review by angarita-dev — CHANGES_REQUESTED** [review]

File: source/views/Extension/Views/AddNFT/hooks/useSteps.jsx [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  const handleChangeSelectedNFT = (token) => () => {
+    setSelectedToken(token);
+    setStep(1);
+  };
+
+  const handleChangeStep = (index) => setStep(index);
+  const handleClose = () => navigator.navigate('home');
+
+  const leftButton = (onClick) => <LinkButton value={t('common.back')} onClick={onClick} startIcon={BackIcon} />;
+  const rightButton = <LinkButton value={t('common.close')} onClick={() => handleClose()} />;
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/hooks/useSteps.jsx]:
```suggestion
  const rightButton = <LinkButton value={t('common.close')} onClick={handleClose} />;
```

File: source/views/Extension/Views/AddNFT/Steps/Step2.jsx [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+          if (nftCollections?.length) {
+            dispatch(setCollections({ collections: nftCollections, principalId }));
+          }
+          dispatch(setCollectionsLoading(false));
+          setLoading(false);
+          handleClose();
+        }
+      );
+    });
+    console.log('salio todo correcto')
```
Comment (angarita-dev) [inline source/views/Extension/Views/AddNFT/Steps/Step2.jsx]:
```suggestion
```
🙅

**Review by angarita-dev — APPROVED** [review]

**Pull request merged**